prow/plugins/approve/approvers2/owners.go

```diff
@@ -67,59 +59,67 @@ type Owners struct {
 
 // NewOwners consturcts a new Owners instance. filenames is the slice of files changed.
 func NewOwners(log *logrus.Entry, filenames []string, r Repo, s int64) Owners {
-	return Owners{filenamesUnfiltered: filenames, filenames: filenames, repo: r, seed: s, log: log}
+	return Owners{filenames: filenames, repo: r, seed: s, log: log}
 }
 
-// GetApprovers returns a map from ownersFiles -> people that are approvers in them
+// GetApprovers returns a map from filenames -> people that can approve it
```

Comment:
Yes this changes the logic to look at each file rather than each directory.

>What is the performance impact of this? e.g. any more github api calls than before?

I did some test on some really large PRs (~1200 files changes, ~500k lines modified) to see if there was any impact. I did not notice any significant slow down on impact when running these tests (I setup a separate prow instance to test this - I dont have those numbers with me now). 

I dont believe this makes any additional github calls. It uses the same information that the approval plugin receives. 

>How would this perform against some of the gnarlier size/XXL PR's we've seen? e.g. would the approvalnotifier comment be truncated?

In the new approval notifier message we only mention the count of the files and the list directories that have file changes. We do not list all the individual files. With this the message size does not grow a lot from the previous implementation and there is no truncation.